Docs: touch responsiveness optimization

patrickhlauke opened the pull request:
Admittedly wordy info dump on why we use `touch-action:manipulation`

refs https://github.com/twbs/bootstrap/pull/18051#issuecomment-156375970

patrickhlauke committed:
Touch responsiveness optimization

Admittedly wordy info dump on why we use `touch-action:manipulation`

refs https://github.com/twbs/bootstrap/pull/18051#issuecomment-156375970

patrickhlauke commented:
@cvrebert and @mdo ... any thoughts on this would be super

cvrebert commented:
* Use the markdown syntax for the hyperlinks
* Use `{{ site.baseurl }}` in the internal links

cvrebert commented on docs/content/reboot.md:
```diff
@@ -301,3 +301,13 @@ HTML5 adds [a new global attribute named `[hidden]`](https://developer.mozilla.o
+
+Most mobile browsers automatically optimize away this 300ms delay for sites that use the `width=device-width` property as part of their <a href="/getting-started/introduction/#responsive-meta-tag">responsive meta tag</a> (as well as for sites that disable zooming, for instance with `user-scalable=no`, though this practice is strongly discouraged for accessibility and usability reasons). The only  ...
+
+On touch-enabled laptop/desktop devices, the majority of browsers don't have any "double-tap to scroll" functionality, and therefore don't have any delay. The exception here are IE11 and Microsoft Edge, which do allow the user to double-tap. As the <a href="/getting-started/introduction/#responsive-meta-tag">responsive meta tag</a> is ignored by desktop browsers (including IE11 and Microsoft Edge ...
```
"The **exceptions** here" (plural)

patrickhlauke commented:
Good catches @cvrebert - fixed.

cvrebert commented on docs/content/reboot.md:
```diff
@@ -301,3 +301,13 @@ HTML5 adds [a new global attribute named `[hidden]`](https://developer.mozilla.o
+
+Most mobile browsers automatically optimize away this 300ms delay for sites that use the `width=device-width` property as part of their [responsive meta tag]({{ site.baseurl }}/getting-started/introduction/#responsive-meta-tag) (as well as for sites that disable zooming, for instance with `user-scalable=no`, though this practice is strongly discouraged for accessibility and usability reasons). Th ...
+
+On touch-enabled laptop/desktop devices, the majority of browsers don't have any "double-tap to scroll" functionality, and therefore don't have any delay. The exceptions here are IE11 and Microsoft Edge, which do allow the user to double-tap. As the [responsive meta tag]({{ site.baseurl }}/getting-started/introduction/#responsive-meta-tag) is ignored by desktop browsers (including IE11 and Micros ...
```
> The exceptions here are IE11 and Microsoft Edge, which do allow the user to double-tap.

I would either add "to scroll" or rephrase to something like "which do implement this feature".

cvrebert replied:
Hyperlink "`touch-action:manipulation` CSS property" to https://developer.mozilla.org/en-US/docs/Web/CSS/touch-action ?

patrickhlauke replied:
don't think the moz page adds any info that isn't in the actual spec at this point?

patrickhlauke replied:
then again, it does have the compat table etc. on second thoughts, yeah, i'll link to that and remove the then redundant parenthetical about the W3C spec.

cvrebert commented:
> though this is likely to change in future versions of iOS

Perhaps say it's presumed to land in iOS 10?

cvrebert commented:
> The only exception here is currently iOS Safari

What about IE Mobile 10-11?

cvrebert commented on docs/content/reboot.md:
```diff
@@ -301,3 +301,13 @@ HTML5 adds [a new global attribute named `[hidden]`](https://developer.mozilla.o
 
 To merely toggle the visibility of an element, meaning its `display` is not modified and the element can still affect the flow of the document, use [the `.invisible` class]({{ site.baseurl }}/components/utilities/#invisible-content) instead.
+
+## Touch responsiveness optimization
```
Perhaps this should use a word other than "responsiveness" to avoid confusion with responsive design?

patrickhlauke replied:
"response" rather than "responsiveness" perhaps? or simply "Touch optimization" (though this may be wrongly be interpreted as a much larger topic, like touch-optimized carousels etc)?

cvrebert replied:
"Click event optimization on touch devices" maybe?

patrickhlauke commented:
> What about IE Mobile 10-11?
By the time v4 goes out of alpha, Windows 10 Mobile will be out (expected ship date of Win 10 Mobile phones is next month) and I'd consider that the most up-to-date version for WinPhone platform. Perhaps changing "only" to "biggest" may solve this without requiring some fiddly rewording (as IE11 does suppress delay for non-zoomable pages)

cvrebert commented:
> This delay is necessary for these browsers to correctly handle "double-tap to scroll" gestures

What about double-tap-to-zoom gestures?

patrickhlauke commented:
> What about double-tap-to-zoom gestures?

doh, of course i meant zoom...it was just an Apple "feature" that they also scroll, which they kept bringing up as a reason not to do the optimizations. this must have embedded itself in my brain

patrickhlauke commented:
Reworded the desktop bit to make it flow better, simplified the iOS Safari part, added mention of IE11/WinPhone, retitled to something hopefully terse enough

cvrebert commented:
> The biggest exceptions here are currently [...] and IE11 on Windows Phone 8.1. These browsers require additional scripts such as [FastClick](https://github.com/ftlabs/fastclick) to explicitly work around the delay.

IE11 on Windows Phone 8.1 doesn't require extra scripts; it supports `touch-action:manipulation`.

cvrebert commented:
Perhaps move the "though this is likely to change" part into the "In the case of iOS" paragraph?

cvrebert commented on docs/content/reboot.md:
```diff
@@ -301,3 +301,17 @@ HTML5 adds [a new global attribute named `[hidden]`](https://developer.mozilla.o
+
+Most mobile browsers automatically optimize away this 300ms delay for sites that use the `width=device-width` property as part of their [responsive meta tag]({{ site.baseurl }}/getting-started/introduction/#responsive-meta-tag) (as well as for sites that disable zooming, for instance with `user-scalable=no`, though this practice is strongly discouraged for accessibility and usability reasons). Th ...
+
+On touch-enabled laptop/desktop devices, IE11 and Microsoft Edge are currently the only browsers with "double-tap to scroll" functionality. As the [responsive meta tag]({{ site.baseurl }}/getting-started/introduction/#responsive-meta-tag) is ignored by all desktop browsers, using `width=device-width` will have no effect on the 300ms delay here.
```
Should be "zoom" instead of "scroll", right?

patrickhlauke replied:
ARGH. note to self: never start writing these things around midnight...you're, of course, correct

patrickhlauke commented:
> Perhaps move the "though this is likely to change" part into the "In the case of iOS" paragraph?

think it makes more sense in the "iOS doesn't support this" para, because that is what's going to change (and the suggested approach will then change as a result of that, of course)

cvrebert commented:
LGTM then.
:shipit:

patrickhlauke commented:
thanks for your patience and keen eye there. :100:

cvrebert commented:
Thanks for saving me from having to write these docs myself :smile:
